pybamm/solvers/base_solver.py

```diff
@@ -640,73 +663,98 @@ def solve(self, model, t_eval=None, external_variables=None, inputs=None):
         # object, restarting the solver at each discontinuity (and recalculating a
         # consistent state afterwards if a dae)
         old_y0 = model.y0
-        solution = None
+        solutions = None
         for start_index, end_index in zip(start_indices, end_indices):
             pybamm.logger.info(
                 "Calling solver for {} < t < {}".format(
                     t_eval_dimensionless[start_index] * model.timescale_eval,
                     t_eval_dimensionless[end_index - 1] * model.timescale_eval,
                 )
             )
-            new_solution = self._integrate(
-                model, t_eval_dimensionless[start_index:end_index], ext_and_inputs
-            )
-            new_solution.solve_time = timer.time()
-            if solution is None:
-                solution = new_solution
+            ninputs = len(ext_and_inputs_list)
+            if ninputs == 1:
```

Comment:
This is the important bit where we solve one model if `inputs` was a dictionary or an ensemble of models using multiprocessing if it was a list of dicts